R/dashboard.Rmd

```diff
@@ -1,453 +1,516 @@
+  
+  tab3 <- fetch(rs3, n=-1)
+    
+  export_cur <- ceiling(tab3$TotalExport_current/1000000)
+}else{
+  export_cur <- ceiling(sum(exportsLatestMonth$Value, na.rm=TRUE)/1000000)
+}   
+
+valueBox(export_cur,
+         icon = 'fa-arrow-right',
+         color = "green")
+```
+
+### Current Month Trade Balance
+
+```{r trade balance value box}
+
+tradebalance_cur <- export_cur - import_cur
+
+valueBox(tradebalance_cur,
+         icon = 'fa-balance-scale',
+         color = "red")
+```
+
+Row
+-------------------------------------
+
+### Services Import/Export Analysis
+
+```{r year value box}
+valueBox(paste(format(params$endDate, "%Y")),
+         color = "warning")
+```
+
+### Total Services Export (`r format(params$endDate, "%Y")`)
+
+```{r total export value box}
+serviceExport <- NULL
+if(params$useSQL){
+  
+  service_export = dbSendQuery(mydb, "SELECT ServiceExport FROM tblservices WHERE Year ='2019'")
+    
+  servicesExport <- fetch(service_export, n=-1)
+  serviceExport <- ceiling(servicesExport$ServiceExport)
+}else{
+  serviceExport <- sum(historicExports[historicExports$Year == format(params$endDate, "%Y"), "Value"], na.rm=TRUE)
+}
+
+
+valueBox(serviceExport,
+         icon = 'fa-arrow-left',
+         color = "blue")
+```
+
+### Total Services Import (2019)
+
+```{r total import value box}
+serviceImport <- NULL
+if(params$useSQL){
+  service_import = dbSendQuery(mydb, "SELECT ServiceImport FROM tblservices WHERE Year ='2019'")
+    
+  servicesImport <- fetch(service_import, n=-1)
+  serviceImport <- ceiling(servicesImport$ServiceImport)
+}else{
+  serviceImport <- sum(historicImports[historicImports$Year == format(params$endDate, "%Y"), "Value"], na.rm=TRUE)
+}
+
+valueBox(serviceImport,
+         icon = 'fa-arrow-right',
+         color = "green")
+```
+
+### Services Trade Balance (2019)
+
+```{r total balance value box}
+
+serviceBalance <- serviceExport - serviceImport
+
+valueBox(serviceBalance,
+         icon = 'fa-balance-scale',
+         color = "red")
+```
+
+
+Row
+-------------------------------
+
+### Goods Export/Import/Balance by Year
+
+```{r goods exports/imports/balance by year interactive plot}
+trades %>%
+  group_by(Year) %>%
+  summarise(Export = sum(Export_mil),
+            Import = sum(Import_mil),
+            Balance = sum(TradeDef_mil), .groups="keep") %>%
+  
+  plot_ly(x = ~Year,
+          y = ~Export,
+          color = "blue",
+          type = 'scatter', 
+          mode = 'lines+markers',
+          name = "Export") %>%
+  
+  add_trace(x = ~Year,
+            y = ~Import,
+            color = "red",
+            name = "Import") %>%
+  
+  add_trace(x = ~Year,
+            y = ~Balance,
+            color = "green",
+            name = "Balance") %>%
+  
+  layout(xaxis = list(title = "Year"), yaxis = list(title = 'Value'))

```

Comment:
Changed to line chart and combined into single panel - user can select line using legend